tests/test_bench.py

```
@@ -0,0 +1,779 @@
+                "reasoning": "x",
+                "criterion_scores": {"left_helpfulness": 3},
+            })
+
+
+class TestPricingSplitBetweenArmAndJudge:
+    """Cross-provider / different-model judge runs must bill arm tokens and
+    judge tokens against their respective per-1M rates. Regression: a single
+    `pricing` field was reused for both, mis-reporting judge spend whenever
+    the judge model differed from the arm model."""
+
+    def _build_ctx(self, *, arm_pricing: dict[str, float], judge_pricing: dict[str, float]):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_build_ctx`

(ANN202)